<h1> Exercise 2.1

In [6]:
data <- read.csv("./data/TrainExer21.csv")
names(data)[1] <- "Observation"

In [8]:
summary(data)

  Observation         Wage          LogWage          Female     
 Min.   :  1.0   Min.   : 32.0   Min.   :3.466   Min.   :0.000  
 1st Qu.:125.8   1st Qu.: 72.0   1st Qu.:4.277   1st Qu.:0.000  
 Median :250.5   Median :100.0   Median :4.605   Median :0.000  
 Mean   :250.5   Mean   :114.9   Mean   :4.641   Mean   :0.368  
 3rd Qu.:375.2   3rd Qu.:144.0   3rd Qu.:4.970   3rd Qu.:1.000  
 Max.   :500.0   Max.   :384.0   Max.   :5.951   Max.   :1.000  
      Age             Educ          Parttime    
 Min.   :20.00   Min.   :1.000   Min.   :0.000  
 1st Qu.:32.00   1st Qu.:1.000   1st Qu.:0.000  
 Median :39.00   Median :2.000   Median :0.000  
 Mean   :40.01   Mean   :2.078   Mean   :0.288  
 3rd Qu.:47.00   3rd Qu.:3.000   3rd Qu.:1.000  
 Max.   :70.00   Max.   :4.000   Max.   :1.000  

<h3>(a) Use dataset TrainExer21 to regress log-wage on a constant and the gender dummy ‘Female’, and check the result presented in Lecture 2.1 that log(Wage) = 4.73 − 0.25Female + e.

In [12]:
fit <- lm(log(Wage) ~ Female, data=data)
sprintf("Regression line: log(Wave) = %.2f - %.2fFemale + e", coef(fit)[1], -1*coef(fit)[2])

[1] "Regression line: log(Wave) = 4.73 - 0.25Female + e"

<h3>(b) Let e be the series of residuals of the regression in part (a). Perform two regressions:  <br>
&nbsp&nbsp&nbsp&nbsp(i) e on a constant and education;<br>
&nbsp&nbsp&nbsp&nbsp(ii) e on a constant and the part-time job dummy.</h3>

In [23]:
data_pred <- data
data_pred$e <- summary(fit)$residuals

residual_fit1 <- lm(e ~ Educ, data=data_pred)
sprintf("Regression line on education: e = %.2f + %.2fEduc", coef(residual_fit1)[1], coef(residual_fit1)[2])
summary(residual_fit1)

[1] "Regression line on education: e = -0.45 + 0.22Educ"


Call:
lm(formula = e ~ Educ, data = data_pred)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.07907 -0.28095 -0.01408  0.25163  1.21331 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) -0.45265    0.03614  -12.53   <2e-16 ***
Educ         0.21783    0.01550   14.05   <2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.3662 on 498 degrees of freedom
Multiple R-squared:  0.2839,	Adjusted R-squared:  0.2824 
F-statistic: 197.4 on 1 and 498 DF,  p-value: < 2.2e-16


In [22]:
residual_fit2 <- lm(e ~ Parttime, data=data_pred)
sprintf("Regression line on education: e = %.2f + %.2fParttime", coef(residual_fit2)[1], -1*-1*coef(residual_fit2)[2])
summary(residual_fit2)

[1] "Regression line on education: e = -0.03 + 0.10Parttime"


Call:
lm(formula = e ~ Parttime, data = data_pred)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.08753 -0.30477 -0.02774  0.30681  1.44257 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)  
(Intercept) -0.02842    0.02281  -1.246   0.2134  
Parttime     0.09869    0.04251   2.322   0.0207 *
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.4304 on 498 degrees of freedom
Multiple R-squared:  0.01071,	Adjusted R-squared:  0.008721 
F-statistic:  5.39 on 1 and 498 DF,  p-value: 0.02065


<h3> (c)  Comment on the outcomes of regressions (i) and (ii) of part (b). </h3>

e has a strong linear relationship with Educ. This means Education can explain e much better than Parttime does. <br><br>

More specifically:<br>
Having highed education adds 22% to the unexplained part of Wage (e)<br>
Having a part-time job adds 10% to the unexplained part of Wage (e)<br>